.github/workflows/bbm_build_crossreference.yml

```diff
@@ -0,0 +1,97 @@
+---
+name: bbm-build-crossreference
+
+on:
+  push:
+    paths:
+      - "cross-reference/**"
+      - .github/workflows/bbm_build_crossreference.yml
+  pull_request:
+    paths:
+      - "cross-reference/**"
+      - .github/workflows/bbm_build_crossreference.yml
+
+defaults:
+  run:
+    working-directory: cross-reference
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: build
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check Dockerfile with hadolint
+        run: |
+          docker run -i -v $(pwd):/mnt -w /mnt ghcr.io/hadolint/hadolint:latest hadolint /mnt/Dockerfile
```

Comment:
Here is how to check the correct Dockerfile:
```diff
- docker run -i -v $(pwd):/mnt -w /mnt ghcr.io/hadolint/hadolint:latest hadolint /mnt/Dockerfile
+ docker run -i -v $(pwd):/mnt -w /mnt ghcr.io/hadolint/hadolint:latest hadolint /mnt/cross-reference/Dockerfile
```

Reply:
This doesn't work because of line 16. All the next steps need to run in the cross-reference dir. So we either make this run in the parent, or copy the file

Reply:
Oh, I see, sorry missed that one. I am happy with any one of the mentioned solution.